experimental/dispatch_profiler/matmul.py

```diff
@@ -0,0 +1,590 @@
+                                         operation.name() + '.mlir')
+
+    # path to iree-compile tool. (for compiling the input mlir file to vmfb)
+    self.iree_compile_path = os.path.join(args.build_dir, 'tools/iree-compile')
+    self.force_compile = False if args.force_compile in ['False', 'false', '0'
+                                                        ] else True
+
+    # path to iree-benchmark-module tool. (for performance benchmarking and profiling)
+    self.iree_benchmark_module_path = os.path.join(
+        args.build_dir, 'tools/iree-benchmark-module')
+
+    # path to iree-run-module tool. (for verification)
+    self.iree_run_module_path = os.path.join(args.build_dir,
+                                             'tools/iree-run-module')
+
+    # output vmfb files for the operation.
+    self.vmfb_verify_file = os.path.join(self.operation_path,
+                                         self.operation.name() + '_verify.vmfb')
+    self.vmfb_benchmark_file = os.path.join(
+        self.operation_path,
+        self.operation.name() + '_benchmark.vmfb')
+
+  def compile(self, compilation_mode):
+    """Compiles the matmul operation to a vmfb file for profiling."""
+    cmd_template = \
+    "${iree_compile_path} ${source_mlir_file} --iree-hal-target-backends=${device} "\
+    "--iree-hal-cuda-llvm-target-arch=sm_80 --iree-hal-benchmark-dispatch-repeat-count=${benchmark-dispatch-repeat-count} "\
+    "--iree-codegen-llvmgpu-use-mma-sync=true -o ${vmfb_file}"
+
+    benchmark_dispatch_repeat_count = self.benchmark_dispatch_repeat_count if compilation_mode == CompilationMode.Benchmark else 1
+    vmfb_file = self.vmfb_benchmark_file if compilation_mode == CompilationMode.Benchmark else self.vmfb_verify_file
+
+    values = {
+        'iree_compile_path': self.iree_compile_path,
+        'source_mlir_file': self.source_mlir_file,
+        'device': self.device,
+        'benchmark-dispatch-repeat-count': str(benchmark_dispatch_repeat_count),
+        'vmfb_file': vmfb_file
+    }
+
+    if not os.path.exists(vmfb_file) or self.force_compile:
+      cmd = SubstituteTemplate(cmd_template, values)
+      print(">> Compilation command for " +
+            CompilationModeNames[compilation_mode] + " : " + cmd)
+      subprocess.getoutput(cmd)
+
+    else:
+      if self.verbose:
+        print("Skipping compilation of matmul operation: " + vmfb_file +
+              " since it already exists.")
+
+  def verify(self, configuration):
+    """Verifies the matmul operation with a given configuration."""
+    # First compile the operation to a vmfb file.
+    self.compile(CompilationMode.Verify)
+
+    # Template for the verification command using iree-run-module as the verification tool.
+    cmd_templete = \
+    "${iree_run_module_path} --module=${vmfb_file} --device=${device} --function=${function_name} "\
+    "--input=@${lhs_npy_file} --input=@${rhs_npy_file} "\
+    "--expected_output=@${expected_result_npy_file}"
```

Comment:
Same here, maybe use a format string?
```suggestion
    cmd_templete = \
    "{iree_run_module_path} --module={vmfb_file} --device={device} --function={function_name} "\
    "--input=@{lhs_npy_file} --input=@{rhs_npy_file} "\
    "--expected_output=@{expected_result_npy_file}".format(
        iree_run_module_path=self.iree_run_module_path,
        vmfb_file=self.vmfb_verify_file,
        device=self.device,
        function_name=function_name,
        lhs_npy_file=lhs_npy_file,
        rhs_npy_file=rhs_npy_file,
        expected_result_npy_file=expected_result_npy_file
    )
```